svc-events/events/subscriptionCache.go

```diff
@@ -0,0 +1,210 @@
+package events
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	l "github.com/ODIM-Project/ODIM/lib-utilities/logs"
+	"github.com/ODIM-Project/ODIM/svc-events/evmodel"
+	uuid "github.com/satori/go.uuid"
+)
+
+var (
+	subscriptionsCache                    = make(map[string]evmodel.Subscription)
+	systemToSubscriptionsMap              = make(map[string]map[string]bool)
+	aggregateIdToSubscriptionsMap         = make(map[string]map[string]bool)
+	collectionToSubscriptionsMap          = make(map[string]map[string]bool)
+	emptyOriginResourceToSubscriptionsMap = make(map[string]map[string]bool)
+	systemIdToAggregateIdsMap             = make(map[string]map[string]bool)
+	eventSourceToManagerIDMap             = make(map[string]string)
+	managerIDToSystemIDsMap               = make(map[string][]string)
+	managerIDToChassisIDsMap              = make(map[string][]string)
+)
+
+func LoadSubscriptionData() {
+	l.Log.Debug("Event data load initialized ")
+	t := time.Now()
+	defer l.Log.Debug("Time take to read Complete LoadSubscriptionData ", time.Since(t))
+	getAllSubscriptions()
+	getAllAggregates()
+	getAllDeviceSubscriptions()
+}
+func getAllSubscriptions() {
+	subscriptionsCache = make(map[string]evmodel.Subscription)
```

Comment:
here again construct the map only after knowing the size. Else we keep extending each of these maps multiple times

Reply:
done